python/tk_houdini/base_hooks/node_handler_base.py

```diff
@@ -0,0 +1,471 @@
+        """
+        raise NotImplementedError("'_restore_sgtk_parms' needs to be implemented")
+
+    def restore_sgtk_parms(self, node):
+        """
+        Restore any removed sgtk parameters onto the given node.
+
+        :param node: A :class:`hou.Node` instance containing sgtk parameters.
+        """
+        sgtk_identifier = node.parm(self.SGTK_IDENTIFIER)
+        use_sgtk = node.parm(self.USE_SGTK)
+        if use_sgtk or not sgtk_identifier:
+            return
+        self._restore_sgtk_parms(node)
+
+    #############################################################################################
+    # houdini callback overrides
+    #############################################################################################
+    
+    def on_created(self, node=None):
+        """
+        Method to run on houdini's OnCreated callback.
+
+        :param node: A :class:`hou.Node` instance.
+        """
+        self.add_sgtk_parms(node)
+
+    def on_deleted(self, node=None):
+        """
+        Method to run on houdini's OnDeleted callback.
+
+        :param node: A :class:`hou.Node` instance.
+        """
+        pass
+
+    def on_input_changed(self, node=None):
+        """
+        Method to run on houdini's OnInputChanged callback.
+
+        :param node: A :class:`hou.Node` instance.
+        """
+        pass
+
+    def on_loaded(self, node=None):
+        """
+        Method to run on houdini's OnLoaded callback.
+
+        :param node: A :class:`hou.Node` instance.
+        """
+        pass
+
+    def on_name_changed(self, node=None):
+        """
+        Method to run on houdini's OnNameChanged callback.
+
+        :param node: A :class:`hou.Node` instance.
+        """
+        pass
+
+    def on_updated(self, node=None):
+        """
+        Method to run on houdini's OnUpdated callback.
+
+        :param node: A :class:`hou.Node` instance.
+        """
+        pass
+
+    def before_first_create(self, node=None):
+        """
+        Method to run on houdini's BeforeFirstCreated callback.
+
+        :param node: A :class:`hou.Node` instance.
+        """
+        pass
+
+    def after_last_delete(self, node=None):
+        """
+        Method to run on houdini's AfterLastDelete callback.
+
+        :param node: A :class:`hou.Node` instance.
+        """
+        pass
+
+    #############################################################################################
+    # UI customisation
+    #############################################################################################
+        
+    def _get_parameter_group(self, node):
+        """
+        Get the given node's parameter group.
+
+        :param node: A :class:`hou.Node` instance.
+
+        :rtype: :class:`ParmGroup`.
+        """
+        if not node:
+            return
+        tk_houdini = self.parent.import_module("tk_houdini")
+        utils = tk_houdini.utils
+        return utils.wrap_node_parameter_group(node)
```

Comment:
Add missing mention of `None` into return type, single exit point

```suggestion
        :rtype: :class:`ParmGroup` or None.
        """
        group = None
        if node:
            tk_houdini = self.parent.import_module("tk_houdini")
            group = tk_houdini.utils.wrap_node_parameter_group(node)
        return group
```